ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/sync/fullsync/FullSyncChainDownloaderTest.java

```diff
@@ -568,8 +565,19 @@ public void requestsCheckpointsFromSyncTarget() {
     int count = 0;
     while (localBlockchain.getChainHeadBlockNumber() < bestPeerChainHead) {
       if (count > 10_000) {
-        fail("Did not reach chain head soon enough");
+        fail(
+            "Did not reach chain head soon enough. At: "
+                + localBlockchain.getChainHeadBlockNumber()
+                + " Chain head: "
+                + bestPeerChainHead);
       }
+      Awaitility.await()
+          .atMost(10, TimeUnit.SECONDS)
+          .until(
+              () ->
+                  bestPeer.hasOutstandingRequests()
+                      || otherPeers.stream().anyMatch(RespondingEthPeer::hasOutstandingRequests)
+                      || localBlockchain.getChainHeadBlockNumber() >= bestPeerChainHead);
```

Comment:
I suspect this test is intermittent in master, but it becomes *really* intermittent once you make `getChainHeadBlockNumber` much faster.  So we wait for there to be an outstanding request to respond to.